pkg/gen/nodejs-templates/yaml.ts.mustache

```diff
@@ -393,26 +393,22 @@ import * as outputs from "../types/output";
             {{/Groups}}
         ) {
             const objs = [];
-            const items = obj["items"] || [];
+            const items = obj.items ?? [];
             for (const item of items) {
                 objs.push(...parseYamlObject(item, transformations, resourcePrefix, opts));
             }
             return objs;
         }
 
-        if (!("metadata" in obj) || !("name" in obj["metadata"])) {
+        if (!("metadata" in obj) || !("name" in obj.metadata)) {
             throw new Error(`YAML object does not have a .metadata.name: ${obj.apiVersion}/${obj.kind} ${JSON.stringify(obj.metadata)}`)
         }
 
-        const meta = obj["metadata"];
-        let id: pulumi.Output<any> = pulumi.output(meta["name"]);
-        const namespace = meta["namespace"] || undefined;
-        if (namespace !== undefined) {
-            id = pulumi.concat(namespace, "/", id);
-        }
-        if (resourcePrefix !== undefined) {
-            id = pulumi.concat(resourcePrefix, "-", id);
-        }
+        const meta = obj.metadata;
+        const name: string = resourcePrefix ? `${resourcePrefix}-${meta.name}` : meta.name;
+        const namespace = meta?.namespace;
+        const id: pulumi.Output<string> = namespace ? pulumi.concat(namespace, "/", name) : pulumi.output<string>(name);
```

Comment:
I don't think that this is equivalent to the original code if both `resourcePrefix` and `namespace` are specified. In the original code, that would result in an ID like:

```
prefix-namespace/name
```

With the new code, we will have

```
namespace/prefix-name
```

Reply:
Ugh, you're right. The rewrite is what I intended originally, but we probably have to keep it now.